### PR TITLE
Download firmware

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,94 +14,84 @@ else
...
+JAG_TOIT_DEPENDENCIES += $(SDK_PATH)/bin/toit.pkg
+JAG_TOIT_DEPENDENCIES += $(JAG_TOIT_REPO_PATH)/build/esp32/firmware.envelope
+
+.PHONY: $(JAG_TOIT_DEPENDENCIES)
+$(JAG_TOIT_DEPENDENCIES): toit-repo-git-tags
+	make -C $(JAG_TOIT_REPO_PATH) version-file esp32
+
+.PHONY: toit-repo-git-tags
+toit-repo-git-tags:
+	(cd $(JAG_TOIT_REPO_PATH); git fetch --tags --recurse-submodules=no)
```
Comment (floitsch) [inline Makefile:96]: I would remove that one.
When we work with a locally checked out toit-repo, I expect that directory to have all submodules checked out, and all tags to be there.

File: cmd/jag/directory/directory.go [hunk 2]
```diff
@@ -138,27 +118,42 @@ func getAssetPath(name string) (string, error) {
...
 	}
 
-	assetsPath, err := GetAssetsPath()
+	assetsPath, err := GetAssetsCachePath()
 	if err != nil {
 		return "", err
 	}
-	snapshotPath := filepath.Join(assetsPath, name)
+	if stat, err := os.Stat(assetsPath); err != nil || !stat.IsDir() {
+		return "", fmt.Errorf("the path '%s' did not hold the Jaguar assets.\nYou must setup the assets using 'jag setup'", assetsPath)
```
Comment (floitsch) [inline cmd/jag/directory/directory.go:129]:
```suggestion
		return "", fmt.Errorf("the path '%s' does not hold the Jaguar assets.\nYou must setup the assets using 'jag setup'", assetsPath)
```

File: cmd/jag/directory/directory.go [hunk 3]
```diff
@@ -138,27 +118,42 @@ func getAssetPath(name string) (string, error) {
...
+	assetsPath, err := GetAssetsPath()
+	if err != nil {
+		return "", nil
+	}
 
-	if stat, err := os.Stat(snapshotPath); err != nil || stat.IsDir() {
-		return "", fmt.Errorf("the path '%s' did not hold the snapshot file.\nYou must setup the Jaguar snapshot using 'jag setup'", snapshotPath)
+	path := filepath.Join(assetsPath, name)
+	if stat, err := os.Stat(path); err != nil || stat.IsDir() {
+		return "", fmt.Errorf("the path '%s' did not hold the asset '%s'.\nYou must setup the Jaguar assets using 'jag setup'", name, path)
```
Comment (floitsch) [inline cmd/jag/directory/directory.go:142]:
```suggestion
		return "", fmt.Errorf("the path '%s' does not hold the asset '%s'.\nYou must setup the Jaguar assets using 'jag setup'", name, path)
```

**Review by floitsch — APPROVED** [review]
LGTM!

https://user-images.githubusercontent.com/1731210/192396428-4acddf32-163b-451c-a535-207d26dcf17a.mp4

File: Makefile (hunk 1 above)
Comment (kasperl) [reply Makefile]: Done.

**Pull request merged**